### PR TITLE
XCB setMouseCursor

### PR DESCRIPTION
**Pull request by fundies**
Adds setMouseCursor() function for linux as per issue #269. Please tell me what to put for doxygen comments and anything else to change.

```
#include <SFML/Graphics.hpp>

int main()
{
    sf::RenderWindow window(sf::VideoMode(800, 600), "My window");
    window.setMouseCursor(142); 

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }
        window.clear(sf::Color::Black);
        window.display();
    }

    return 0;
}
```

### CODE REVIEW

**Comment by eXpl0it3r** [comment]
Can you please squash you two commits into one and use a proper commit message?

The discussion on this PR or the feature itself can be found on [the forum](http://en.sfml-dev.org/forums/index.php?topic=17299.0).

**Comment by fundies** [comment]
No idea how to do either of those things.

**Comment by eXpl0it3r** [comment]
http://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by fundies** [comment]
Well, I only made things worse. I give up.

**Comment by Mischa-Alff** [comment]
Your implementation uses the built-in cursor font instead of the themed font. You should look into how to use the themed font, or simply use `xcb-cursor`.

Also, what does `142` mean? You should have an enum or something with the common cursor types.

**Comment by fundies** [comment]
I updated it to use xcb-cursor. Also I implemented setting cursor to image

xcb-cursor takes strings so new syntax is:
```
window.setMouseCursor("watch");
```
For images:
```
 sf::Image cursor;
	if (cursor.loadFromFile("pencil_1.png"))
		window.setMouseCursor(cursor.getSize().x, cursor.getSize().y, cursor.getPixelsPtr());
```

I dunno if i got all the tabs. I can't tell. Still no idea where / how I put / make enum so any guidence on that would be appricated.

File: src/SFML/Window/Unix/WindowImplX11.hpp [hunk 1]
```diff
@@ -228,19 +246,23 @@ class WindowImplX11 : public WindowImpl
...
+    ::Window                    m_window;              ///< X11 structure defining our window
+    ::Display*                  m_display;             ///< Pointer to the display
+    xcb_connection_t*           m_connection;          ///< Pointer to the xcb connection
+    xcb_screen_t*               m_screen;              ///< Screen identifier
+    XIM                         m_inputMethod;         ///< Input method linked to the X display
+    XIC                         m_inputContext;        ///< Input context used to get unicode input in our window
+    bool                        m_isExternal;          ///< Tell whether the window has been created externally or by SFML
+    Atom                        m_atomClose;           ///< Atom used to identify the close event
+    int                         m_oldVideoMode;        ///< Video mode in use before we switch to fullscreen
+    bool						m_cursorVisible;       ///< Is the cursor visible
```
Comment (select-this) [inline src/SFML/Window/Unix/WindowImplX11.hpp]: Still got some tabs here, but other than that they all seem to be gone.

**Comment by fundies** [comment]
Windows and Linux now work. I'm having trouble figuring out os x...

**Comment by binary1248** [comment]
Superseded by #827.

**Pull request closed without merging**